pilot/pkg/config/kube/gateway/testdata/route-binding.yaml

```diff
@@ -72,6 +74,23 @@ spec:
     - name: httpbin
       port: 80
 ---
+apiVersion: networking.istio.io/v1beta1
+kind: VirtualService
+metadata:
+  name: httpbin-headers
+  namespace: default
+spec:
+  hosts:
+  - "beta.foobar.example"
+  gateways:
+  - "istio-system/gateway-istio-autogenerated-k8s-gateway-foobar"
```

Comment:
I thought we were doing `gateway.networking.k8s.io:example-gateway`? I definitely do not think we should expose the auto-generated stuff

Reply:
This is just for testing. The idea is that the user would set the gateway to `gateway.networking.k8s.io:example-gateway.foobar` which will be converted by [resolveGatewayName](https://github.com/istio/istio/blob/39a1119fd20ff18c237ea76e3f9c805b155764b4/pilot/pkg/model/config.go#L262) to the internal name before this tested code is hit.

I suppose that even though it won't be documented, users might still point a VS to an internal gateway, like they can today. Maybe we could add another validation check to disallow that?

Reply:
^^^^ @howardjohn WDYT about disallowing gateway internal gateway refs in the validator? Or maybe `istioctl analyze` could warn?

Reply:
Seems that `istioctl analyze` already produces errors and warnings when a VS is pointing to an internal GW:
```
Error [IST0101] (VirtualService sample-ingress/httpbin-headers) Referenced gateway not found: "sample-ingress/example-gateway-istio-autogenerated-k8s-gateway-default"
Warning [IST0132] (VirtualService sample-ingress/httpbin-headers) one or more host [httpbin.example.com] defined in VirtualService sample-ingress/httpbin-headers not found in Gateway sample-ingress/example-gateway-istio-autogenerated-k8s-gateway-default.
```
These messages are sort of misleading, since the VS does actually work as expected.

I think it would be better to add a new IST0XXX error code about relying on internal implementation details, and use it instead. The suggested error resolution in the docs is to use `gateway.networking.k8s.io:example-gateway` for the reference instead.

We could also add another IST0XXX warning for when `gateway.networking.k8s.io:example-gateway` syntax is used to suggest that proper Gateway API Routes should be used instead, if possible.

Reply:
I think that is basically a bug in the analyzer since it doesn't know the internal Gateway types

Reply:
Yeah, I updated the analyzer to handle Gateway API better. PTAL https://github.com/istio/istio/pull/47101/files#diff-554352f12aeaaa9495a25770c898d52c7f4f24a532ee7d80d817154b1cf5c441